implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/DynatraceConfig.java

```diff
@@ -129,7 +137,7 @@ default boolean enrichWithDynatraceMetadata() {
                                                 return checkAll(this,
                                                         checkRequired("apiToken", DynatraceConfig::apiToken),
                                                         checkRequired("uri", DynatraceConfig::uri),
-                                                        checkRequired("deviceId", DynatraceConfig::deviceId),
+                                                        check("deviceId", DynatraceConfig::deviceId).andThen(Validated::nonBlank),
```

Comment:
We can make more complex validation rules to ensure we have required configuration set for v1 or v2. See, for example, the InfluxConfig validation where it also supports v1/v2 and defaults to one or the other depending on other configuration being set.
https://github.com/micrometer-metrics/micrometer/blob/83b7b8c323e741f9e469354afb9843b5044a9ed6/implementations/micrometer-registry-influx/src/main/java/io/micrometer/influx/InfluxConfig.java#L219-L221

Reply:
Oh. Perils of only looking at the diff in the GitHub UI. Ignore me, then.